### DISCUSSION TITLE
New user of MRS_UAV_System (coordinates set file or waypoints)

### DISCUSSION THREAD

**Discussion by umaradeel** [ctu-mrs/mrs_uav_system#135] [Q&A]
Hello,
I am a new user and recently installed Mrs_UAV_system and working on drones, I have installed successfully and tired to run the One Drone GPS  and visualized in the Gazebo. I have sucessfully tested some examples present in the "Example_Tmux_Scripts"
I have major concerns,
 
1. May I know which one is the specific file where you have defined the trajectory coordinates or waypoints like x,y and z?
2. I am trying to provide link of a file which contains different coordinates or waypoints and if I run it should follow the path automatically defined in a separate file? so in short rather than providing specific coordinates or waypoints in the ROS, I want to make it automatic. whenever I run the ROS the drone should follow the trajectory which must be in a separate file.

Looking forward to get directions
Regards

**Comment by DanHert** [comment]
Hello, to command the UAV to a position or a trajectory, use the provided topics/services - https://ctu-mrs.github.io/docs/system/uav_ros_interface.html
If you want to load a trajectory from a file, use the trajectory loader - https://github.com/ctu-mrs/trajectory_loader (it is a part of uav_modules metapackage - https://github.com/ctu-mrs/uav_modules). Following the readme for trajectory loader, you can easily set the simulation up so that the trajectory is flown automatically.

Dan H.

**Reply by umaradeel** [reply]
hello,
i tried to follow the steps but i am getting error(please see the [image), i dont know how to integrated trajectory like "circle_10_5" with drone manually.
1. Should i launch the Gazebo simulator, then spawning drones, running the control core for each drone, arming motors and let drones taking off, and the you can command drones giving waypoints to reach? https://ctu-mrs.github.io/docs/simulation/howto.html

![Screenshot from 2022-03-03 16-27-13](https://user-images.githubusercontent.com/85783902/160587932-17bb2915-219e-4ffc-b1b2-0a931dd4b308.png)

2. Can I assign this trajectory like "circle_10_5" to the existing example "one_drone_gps" ? how can i run this one_drone_gps with different trajectories? what are the changes required, please tell me the steps.

I shall be thankful for a prompt response, as I am new and trying to understand the whole systems so i need your help regarding this matter.
thanks

**Comment by petrapa6** [comment]
This question led me to write a quick crossroad on loading and generating trajectories.

@klaxalk Please, check if all is fine in [trajectories.md](https://ctu-mrs.github.io/docs/system/trajectories.html).

**Reply by klaxalk** [reply]
If you could add some info about what is a trajectory... I am afraid that most people do not distinguish trajectory from a path. We should then write a similar page about loading paths.
